articles/cryptocurrency-tracking-telegram-bot/index.md

```diff
@@ -0,0 +1,189 @@
+
+![](/engineering-education/cryptocurrency-tracking-telegram-bot/ngcq6rvjxrpoibsspq9g.png)
+
+Cryptocurrencies are digital forms of currencies in which transactions are verified and records maintained by a decentralized system using cryptography rather than a centralized authority. Examples of cryptocurrency include Bitcoin (BTC), Ethereum (ETH), and Litecoin (LTC).
+
+### Building a Cryptocurrency Tracker
+
+Cryptocurrencies are volatile, so their prices change every second. There are also many cryptocurrencies in existence, and it can be troublesome to keep track of everyone. A cryptocurrency tracker aggregates multiple coins' prices in a single place and updates itself as their prices change.
+
+This article will build a cryptocurrency tracker by consuming data provided by [CryptoCompare API](https://min-api.cryptocompare.com) using Python.
+
+First, we need to install the [Requests](https://requests.readthedocs.io/en/master/) library, an elegant and simple HTTP library for Python. Requests allow us to send HTTP/1.1 requests too easily by saving us the stress of manually adding query strings to our URLs or form-encode POST data.
+
+In the terminal, type:
+
+```
+pip install requests
+```
+
+Second, we need to write code responsible for sending requests to CryptoCompare to fetch the current prices of currencies we wish to track. CryptoCompare provides an endpoint that returns the data we require so that we will be consuming that.
+
+Create a file named `tracker.py` and save the following code in it:
+
+```Python
+import requests
+
+
+def get_prices():
+    coins = ["BTC", "ETH", "XRP", "LTC", "BCH", "ADA", "DOT", "LINK", "BNB", "XLM"]
+
+    crypto_data = requests.get(
+        "https://min-api.cryptocompare.com/data/pricemultifull?fsyms={}&tsyms=USD".format(",".join(coins))).json()["RAW"]
+
+    data = {}
+    for i in crypto_data:
+        data[i] = {
+            "coin": i,
+            "price": crypto_data[i]["USD"]["PRICE"],
+            "change_day": crypto_data[i]["USD"]["CHANGEPCT24HOUR"],
+            "change_hour": crypto_data[i]["USD"]["CHANGEPCTHOUR"]
+        }
+
+    return data
+```
+
+This is the expected output from the code above when executed:
+
+![](/engineering-education/cryptocurrency-tracking-telegram-bot/2en1wn0m-8gcycfjmtvs.png)
+
+## Building a Telegram Bot
+
+#### Step 1: Have a Conversation with BotFather
+
+Sign In to your Telegram account, then search for `@botfather` and start a conversation with the account. PS: BotFather is also a Telegram bot.
+
+![](/engineering-education/cryptocurrency-tracking-telegram-bot/2frh-rcjna_zv5g1epn7.png)
```

Comment:
![alt text here](image url). This is the format for inserting an image. Can you insert texts describing the image in 2-3 words in the space allotted for it? Do it for all the images.

Reply:
Sure no problem, I've added the alt tag to all images